pulsar-broker/src/main/java/org/apache/pulsar/broker/loadbalance/impl/LinuxBrokerHostUsageImpl.java

```diff
@@ -75,18 +75,17 @@ public LinuxBrokerHostUsageImpl(int hostUsageCheckIntervalMin,
         this.lastCollection = 0L;
         this.usage = new SystemResourceUsage();
         this.overrideBrokerNicSpeedGbps = overrideBrokerNicSpeedGbps;
-        executorService.scheduleAtFixedRate(this::calculateBrokerHostUsage, 0,
-                hostUsageCheckIntervalMin, TimeUnit.MINUTES);
 
         boolean isCGroupsEnabled = false;
         try {
              isCGroupsEnabled = Files.exists(Paths.get(CGROUPS_CPU_USAGE_PATH));
         } catch (Exception e) {
             log.warn("Failed to check cgroup CPU usage file: {}", e.getMessage());
         }
-
         this.isCGroupsEnabled = isCGroupsEnabled;
-        calculateBrokerHostUsage();
```

Comment:
I don't think that the call to `calculateBrokerHostUsage` should be removed. It should remain so that the values are available after the constructor returns. Instead of removing this line, the scheduling should happen after this call and the initial delay of the scheduling should be set to `hostUsageCheckIntervalMin` to prevent the race. 

The problem in the race should also be fixed, I have created a few commits in https://github.com/apache/pulsar/compare/master...lhotari:issue-6368-flaky-test to fix the issue that the concurrent calls to `calculateBrokerHostUsage` caused.

Some suggested improvements to de-risk actual production flakiness:
* [Make ResourceUsage class immutable](https://github.com/lhotari/pulsar/commit/7c5ea55fffd49a2b09b53ca0b1132eef0b1e67cf)
* [Replace usage of Files.readAllBytes with Files.copy](https://github.com/lhotari/pulsar/commit/97a3fb609b6aea81327146d2296412c5599988b2)
* [Pre-calculate the usage percentage in ResourceUsage class](https://github.com/lhotari/pulsar/commit/32aec9de38674f959d090305cb40b14f8c994df7)
* [Fix initial cpu usage calculation and use System.nanoTime() for calculating elapsed time](https://github.com/apache/pulsar/commit/468550f471e3e1b4e5071fa9e5fac2f69713a398)

One possibility is to cherry-pick these commits to this same PR.

Reply:
Great point. I agree that we shouldn't return until the class is fully initialized. I'm guessing that was the original intent of including that method call there, too. I can update the constructor as you suggested, and I think it likely makes the most sense to cherry-pick your commits.

I also noticed that there is a race condition in the `GenericBrokerHostUsageImpl`, where two different scheduled methods make calls that update `cpuUsageSum` and `cpuUsageCount`. I'm wondering if we should add logic to `synchronize` on one of the variables? If you have another suggestion, happy to look into it.

Reply:
> I'm wondering if we should add logic to synchronize on one of the variables? If you have another suggestion, happy to look into it.

Good catch. Yes, using `synchronized` is fine there. 

In `GenericBrokerHostUsageImpl`, it could be worth setting `initialDelay` to the the scheduling period and call the methods in the constructor thread once to ensure that initialization happens in the constructor.

Reply:
Done. Mind taking another look?

Reply:
thanks @michaeljmarshall . LGTM. I'll run the tests I have just to verify that the flakiness is fixed.